yapapi/rest/market.py

```diff
@@ -66,36 +64,25 @@ async def confirm(self) -> bool:
 
         :return: True if the agreement has been confirmed, False otherwise
         """
-        await self._api.confirm_agreement(self._id)
         try:
+            await self._api.confirm_agreement(self._id)
```

Comment:
```suggestion
            await self._api.confirm_agreement(self._id)
        except (ApiException, asyncio.TimeoutError):
            logger.debug("confirmAgreement(%s) failed", self._id, exc_info=True)
        try:
    ```

Reply:
@filipgolem agreed to resolve this without committing